bundles/org.openhab.binding.dirigera/src/main/java/org/openhab/binding/dirigera/internal/handler/scene/SceneHandler.java

```diff
@@ -0,0 +1,127 @@
+    private TimeZoneProvider timeZoneProvider;
+    private Instant lastTrigger = Instant.MAX;
+    private int undoDuration = 30;
+
+    public SceneHandler(Thing thing, Map<String, String> mapping, TimeZoneProvider timeZoneProvider) {
+        super(thing, mapping);
+        super.setChildHandler(this);
+        this.timeZoneProvider = timeZoneProvider;
+        logger.warn("TimeZone {}", this.timeZoneProvider.getTimeZone());
+        // no link support for Scenes
+        hardLinks = Arrays.asList();
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        if (super.checkHandler()) {
+            JSONObject values = gateway().api().readScene(config.id);
+            handleUpdate(values);
+
+            if (values.isEmpty()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE,
+                        "@text/dirigera.scene.status.scene-not-found");
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+            // check if different undo duration is configured
+            if (values.has("undoAllowedDuration")) {
+                undoDuration = values.getInt("undoAllowedDuration");
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+        if (CHANNEL_TRIGGER.equals(channelUID.getIdWithoutGroup())) {
+            if (command instanceof DecimalType decimal) {
+                int commandNumber = decimal.intValue();
+                switch (commandNumber) {
+                    case 0:
+                        gateway().api().triggerScene(config.id, "trigger");
+                        lastTrigger = Instant.now();
+                        scheduler.schedule(this::countDown, 1, TimeUnit.SECONDS);
+                        break;
+                    case 1:
+                        gateway().api().triggerScene(config.id, "undo");
+                        lastTrigger = Instant.MAX;
+                        updateState(new ChannelUID(thing.getUID(), CHANNEL_TRIGGER), UnDefType.UNDEF);
+                        break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public void handleUpdate(JSONObject update) {
+        super.handleUpdate(update);
+        if (update.has("lastTriggered")) {
+            Instant lastRiggeredInstant = Instant.parse(update.getString("lastTriggered"));
+            logger.warn("LastTrigger Instant       {}", lastRiggeredInstant);
+            ZonedDateTime lastTriggeredZDT = lastRiggeredInstant.atZone(timeZoneProvider.getTimeZone());
+            logger.warn("LastTrigger ZonedDateTime {}", lastTriggeredZDT);
+            DateTimeType dtt = new DateTimeType(lastTriggeredZDT);
+            logger.warn("LastTrigger DateTimeType  {}", dtt);
+            updateState(new ChannelUID(thing.getUID(), CHANNEL_LAST_TRIGGER), dtt);
```

Comment:
I can see that you are trying to figure out something related to `DateTimeType` and time-zone. If you are on M5 or the latest snapshot versions after that, you are probably experiencing the effect of openhab/openhab-core#3583. `DateTimeType` no longer contains time-zone information.

So as a result, in line 112 you are calling `DateTimeType.toString()` which will apply system time-zone to the internally stored `Instant`. That might not be the same as your configured time-zone that you obtain from `TimeZoneProvider`.

Your state update can be simplified to:
```suggestion
            Instant lastRiggeredInstant = Instant.parse(update.getString("lastTriggered"));
            DateTimeType dtt = new DateTimeType(lastRiggeredInstant);
            updateState(new ChannelUID(thing.getUID(), CHANNEL_LAST_TRIGGER), dtt);
```

See also #17725 for plenty of examples how this simplifies things in bindings.

Reply:
Done